cat-launcher/src-tauri/src/settings/list_fonts.rs

```diff
@@ -0,0 +1,126 @@
+use std::io;
+use std::path::{Path, PathBuf};
+
+use strum::{IntoEnumIterator, IntoStaticStr};
+
+use crate::filesystem::paths::{
+  get_or_create_user_game_data_dir, get_system_font_dirs,
+  GetUserGameDataDirError,
+};
+use crate::infra::utils::{OSNotSupportedError, OS};
+use crate::settings::Font;
+use crate::variants::GameVariant;
+
+#[derive(thiserror::Error, Debug, IntoStaticStr)]
+pub enum ListFontsError {
+  #[error("failed to get OS: {0}")]
+  OSNotSupported(#[from] OSNotSupportedError),
+
+  #[error("failed to read directory: {0}")]
+  ReadDirectory(#[from] io::Error),
+
+  #[error("failed to get user game data dir: {0}")]
+  UserGameDataDirectory(#[from] GetUserGameDataDirError),
+}
+
+pub async fn list_fonts(
+  data_dir: &Path,
+  os: &OS,
+) -> Result<Vec<Font>, ListFontsError> {
+  let mut fonts = Vec::new();
+
+  // Get fonts from user game data dir (check all variants)
+  for variant in GameVariant::iter() {
+    let user_game_data_dir =
+      get_or_create_user_game_data_dir(&variant, data_dir).await?;
+    let user_font_dir = user_game_data_dir.join("font");
+    if user_font_dir.exists() {
+      fonts.extend(collect_fonts_from_dir(user_font_dir).await?);
+    }
+  }
+
+  // Get fonts from system font directories
+  let system_font_dirs = get_system_font_dirs(os);
+  for dir in system_font_dirs {
+    if dir.exists() {
+      fonts.extend(collect_fonts_from_dir(dir).await?);
+    }
+  }
+
+  // Remove duplicates based on font name
+  let mut seen = std::collections::HashSet::new();
+  fonts.retain(|font| seen.insert(font.name.clone()));
+
+  Ok(fonts)
+}
+
+fn collect_fonts_from_dir(
+  dir: PathBuf,
+) -> std::pin::Pin<
+  Box<
+    dyn std::future::Future<Output = Result<Vec<Font>, io::Error>>
+      + Send,
+  >,
+> {
+  Box::pin(async move {
+    let mut fonts = Vec::new();
+
+    let mut entries = tokio::fs::read_dir(dir).await?;
+    while let Some(entry) = entries.next_entry().await? {
+      let path = entry.path();
+      if !path.is_file() {
```

Comment:
In cat-launcher/src-tauri/src/settings/list_fonts.rs around lines 71 to 74, the
code currently uses the synchronous path.is_file() inside an async function
which can block the runtime; replace that check with an async filesystem call
(use tokio::fs::metadata(path).await?) and then call metadata.is_file() to
determine if the path is a file or directory, propagate or map errors
appropriately, and keep the existing logic (call collect_fonts_from_dir for
directories) while avoiding any blocking sync fs APIs.